pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -593,20 +595,59 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 
 		appInstance.CloudInitUserData = userData
 		appInstance.RemoteConsole = cfgApp.GetRemoteConsole()
+		appInstanceList = append(appInstanceList, appInstance)
+	}
+
+	// validate app instances configuration
+	validateAppInstanceConfig(appInstanceList)
+
+	// publish the config objects
+	for _, appInstance := range appInstanceList {
+		uuidStr := appInstance.UUIDandVersion.UUID.String()
+		// write to zedmanager config directory
+		publishAppInstanceConfig(getconfigCtx, *appInstance)
 		// get the certs for image sha verification
 		certInstance := getCertObjects(appInstance.UUIDandVersion,
 			appInstance.ConfigSha256, appInstance.StorageConfigList)
-
-		// write to zedmanager config directory
-		uuidStr := cfgApp.Uuidandversion.Uuid
-		publishAppInstanceConfig(getconfigCtx, appInstance)
 		if certInstance != nil {
 			publishCertObjConfig(getconfigCtx, certInstance,
 				uuidStr)
 		}
 	}
 }
 
+func validateAppInstanceConfig(appInstanceList []*types.AppInstanceConfig) {
```

Comment:
This code shouldn't be in the zedagent parser but in the zedmanager handling of the appInstanceConfig so that it can report the errors back in AppInstanceStatus

Reply:
Currently, zedmanager does not have the full context, to catch the config parsing errors. I have seen the mechanism inside zedagent, wherein we  are encoding any config parsing errors into the config blobs. However, the corresponding handling inside zedmanager or, other agents are missing. Wherein, when an agent sees an error encoded config blob, should copy the error into the status blob, and stop further processing. I will add this mechanism into zedmanager /zedrouter to start with.

Reply:
What context is missing?

zedmanager and/or zedrouter knows about the networking. zedrouter might be a better place.
But I see no reason to add any code like this to the parser. That just makes the system so much more complex.

Parser should only deal with parsing errors (such as going from string to an net.IP). Nothing more.